auction-server/src/opportunity/api.rs

```diff
@@ -102,148 +183,189 @@ pub struct OpportunityParamsV1 {
     #[serde(with = "crate::serde::u256")]
     pub target_call_value: U256,
 
-    pub sell_tokens: Vec<TokenAmount>,
-    pub buy_tokens:  Vec<TokenAmount>,
+    pub sell_tokens: Vec<TokenAmountEvm>,
+    pub buy_tokens:  Vec<TokenAmountEvm>,
+}
+
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug, ToResponse)]
+#[serde(tag = "version")]
+pub enum OpportunityCreateEvm {
+    #[serde(rename = "v1")]
+    V1(OpportunityCreateV1Evm),
+}
+
+#[derive(Serialize, Deserialize, ToSchema, Clone, ToResponse)]
+pub struct OpportunityEvm {
+    /// The opportunity unique id
+    #[schema(example = "obo3ee3e-58cc-4372-a567-0e02b2c3d479", value_type = String)]
+    pub opportunity_id: OpportunityId,
+    /// Creation time of the opportunity (in microseconds since the Unix epoch)
+    #[schema(example = 1_700_000_000_000_000i128, value_type = i128)]
+    pub creation_time:  UnixTimestampMicros,
+    #[serde(flatten)]
+    // expands params into component fields in the generated client schemas
+    #[schema(inline)]
+    pub params:         OpportunityCreateEvm,
```

Comment:
I think `OpportunityParamsEvm` is a more descriptive type name here? since this `Opportunity...` object may be accessed outside of just creation?